### ISSUE TITLE
[BUG] test.step() with polling inside is shown as failed in html reporting despite it's passed

### ISSUE THREAD

**Issue by Sysqwerty** [microsoft/playwright#23302]
### System info
- Playwright Version: [v1.34.0]
- Operating System: [All]
- Browser: [All]

### Source code
```
public async waitWidgetsLoaded(seconds: number) {
        const loadingDots = await this.widgetRefreshOverlay.all();
        for (const dot of loadingDots) {
            await dot.waitFor({ state: 'hidden', timeout: seconds * 1000 });
        }
        for (const dot of loadingDots) {
            await expect(async () => {
                const isDotContainsAttribute = (await dot.getAttribute('class'))!.includes('widget-loading');
                expect(isDotContainsAttribute).toBeFalsy();
            }).toPass({ timeout: seconds * 1000, intervals: [0.1 * 1000] });
        }
    }
```

**Steps**
- [Run the test] that contains step with polling

Note: polling triggers several times in period of time (condition **is passed** and it's passed **not on 1st polling call**)

**Expected**
Step with polling is marked as **passed** (Green checkmark) in the html report

**Actual**
Step with polling is marked as **failed** (Red checkmark) in the html report

![image](https://github.com/microsoft/playwright/assets/13851264/c0e5447a-65b9-4703-a3d6-d21004c4ad07)

**Comment by aslushnikov** [comment]
I can repro:

```ts
// a.spec.ts
import { test, expect } from '@playwright/test';

test('should work', async () => {
  await test.step('foo', async () => {
    let i = 0;
    await expect(() => {
      expect(i++).toBe(2);
    }).toPass();
  });
});
```

run with:

```
$ npx playwright test --trace on
$ npx playwright show-trace test-results/a-should-work/trace.zip
```

<img width="1392" alt="image" src="https://github.com/microsoft/playwright/assets/746130/11480954-be6e-4ace-a7b4-0d42427d1b36">

**Comment by Sysqwerty** [comment]
Thanks Pavel

**Comment by Sysqwerty** [comment]
The fix will appear from next version [v1.35] ?

**Comment by aslushnikov** [comment]
@Sysqwerty that's correct
